### PR TITLE
Close #6658: Add CODE_OF_CONDUCT

### LINKED ISSUE #6658: Adding Contributor Covenant Code of Conduct
Hey maintainers: Can the project has Contributor Covenant Code of Conduct added https://en.wikipedia.org/wiki/Contributor_Covenant?

### PR DESCRIPTION
**Pull request by tk0miya**
### Purpose
- As proposed in #6658 , I'd like to adopt Contributor Covenant: https://www.contributor-covenant.org

### CODE REVIEW

**Review by TimKam — COMMENTED** [review]
I'm all for it. Do we want to link it in the README and/or docs?

**Review by jfbu — COMMENTED** [review]
-1

I am against it and I can explain why if asked.

I will not object to inclusion in the project, but definitely will not consider such inclusion as a "personal pledge" to such a vague and feel-good text.

I prefer to invite people to study old texts such as the Déclaration des droits de l'homme et du citoyen from the French revolution period.

**Comment by stephenfin** [comment]
+0. Personally, I don't think a document like this will do anything to improve the Sphinx community, which is already a very pleasant one to work in. I struggle to see how covenants like this do anything to resolve real issues they purport to address, mostly in areas like harassment and discrimination, and what's more they distract from the many other issues open source communities face, chief among them being the lack of funding from large commercial entities benefiting from the work of mostly (mostly!) volunteers and giving little back. I won't block it, but I think the fact that there are *maintainers* from Japan, France, Ireland and more already says more about us that this document ever could.

**Comment by TimKam** [comment]
I can understand the meta-level concerns raised here and also agree that mostly, the actions of a community speak for themselves. However, having a CoC signals at first glance that the Sphinx community _intents_ to be a welcoming community and from a pragmatic perspective, I think this is a good sign.

**Review by kynzie — CHANGES_REQUESTED** [review]
-1

If we look at it from a law perspective, that kind of a law is produced to force a social evolution (like `la convention des droits de l'homme et du citoyen` during the French revolution as @jfbu said). Is there anything to force here ?
On another perspective, do we need a text of good character to be polite and respectful ? 
I do not think so. Or the world will end badly.

**Comment by tapaswenipathak** [comment]
Hey folks: How are things happening w/ the document? I would love reading the final document.

**Comment by jfbu** [comment]
I am adding a comment to make more precise my opinion. I feel very much in sync with @kynzie comment. That is, we live in a complex world which has its (frightening) darknesses. The well-wishing text referred too uses strong demands, particularly the world "pledge". I am not a native English speaker, but for me this is a very strong word. But the text itself looks as if written by 12 years old as a school project. For example 

> Project maintainers who do not follow or enforce the Code of Conduct in good faith may face temporary or permanent repercussions as determined by other members of the project’s leadership.

is exactly the kind of thing we expect to see under established dictatorship like Stalin's. As @kynzie indicates what does rather matter are the established rules of law. Almost everything which this code of conduct says should be repressed seems to me to fall under already existing laws. If not, then do some militantism to add additional laws to the society in general.

Concretely, what is happening is that projects via "pledging" to such codes of conduct buy themselves for cheap a good conscience. But at the same time the software industry own habits often proved in the past and present a complete disregard for ethics, as is demonstrated by commercial habits or nowadays by the matters of privacy (or rather lack thereof) and handling of big data for financial gains. That many projects are endorsing such a pledge to a vague "code of conduct" of boy-girl-neither-scout camp level is no excuse for joining the fray.

Personally, as I said "pledging" is a matter of life or death. If I pledge to take care of my ageing mother it has a meaning. I don't think the meaning given to pledges of this kind by the software community at large is as strong. In fact I am 100% convinced it amounts to almost little apart political correctness.

For these reasons, i explicitly state  here that in case Sphinx project adds adhesion to this text, this will not include my personal approval, and I that I personally do not pledge any commitment to https://www.contributor-covenant.org

**Comment by TimKam** [comment]
I suppose the question is now:

Could we potentially agree on any other Code of Conduct that is (from my perspective) similar in intention but better written?

IMHO, a CoC in the context of an Open Source project like Sphinx is mainly a statement that we condemn discrimination (based on gender, religion, ethnicity, etc) and do what we reasonably can to avoid discrimination in the context of the project. This may be an obvious social norm, but there's plenty of empirical evidence that reminding people of such norms helps them adhere to them. That's why I am _for_ a CoC.

**Comment by jfbu** [comment]
I am not against a CoC. I am all for a well written one. It does not hurt indeed to state that Sphinx project obides the rules of law and even beyond. I am sure we are among people of good company and goodwill.

(I will be offline for 10 days; thus inactivity of my part is not to taken as rejection of a proposed CoC, should one be done during next few days)

**Comment by TimKam** [comment]
I think the Django CoC (which is based on a widely used template) seems to be a better fit:
https://www.djangoproject.com/conduct/.
Perhaps we can use it as a starting point? My main motivation for suggesting it is that the Django CoC uses language that is less strong and--from a practical perspective--closer to the truth.

**Comment by tk0miya** [comment]
Sorry for late response. I was also away from network in last 3 weeks.

I think following CoC lets new developers and contributors know this project respects other developers each other. I don't think the text (or wording) of www.contributor-covenant.org itself is important. So django's CoC is also okay to me.
Happily, our development has been respectful each other for long time (thanks all!). So I believe this will not change nothing to us.

**Comment by techtonik** [comment]
>  This may be an obvious social norm, but there's plenty of empirical evidence that reminding people of such norms helps them adhere to them.

I am highly interested to read some scientific papers on that. In my empirical observations licenses and rules serve as a fitness function, a "filter" that removes people, not a "converter" that makes them better. Only people can make other people better, and rules are usually a sign of a deep conflicts that can not be resolved any other way. It gets even more interesting if we add influence/political technologies to the mix. I've seen some really hot debates over that stuff in some open source communities, and I'd say some communities really need a good research in this area to stay sane.

**Comment by TimKam** [comment]
@techtonik [Here](https://onlinelibrary.wiley.com/doi/pdf/10.1002/%28SICI%291520-6793%28199910%2916%3A7%3C545%3A%3AAID-MAR1%3E3.0.CO%3B2-I?casa_token=G3QrgJqqLLwAAAAA:CjcjPsnWNoDe1ixz7bgULxbbrLQNbTEf62iz1lY_ZVNpq3pSsp8lMui_bxBj1JfurGtpr8893Eh8HjVR) is an example of a study that successfully *primes people to behave fairly*. Of course, it is possible to question individual studies, but generally, you can find many studies that show that priming works. I agree there are a lot of deep, philosophical implications that a CoC cannot address. Still, from my perspective, the bigger problem would be that we cannot agree on a statement that essentially says "Let's all be reasonable and nice people". IMHO, we should not *over-engineer* the CoC :-) I'll try to add the Django-style CoC to the PR.

**Comment by jfbu** [comment]
I think reaction on this topic highly depens on whether one has experienced or not living in an authoritarian society such as former communist countries, or currently China and other countries. In particular I am convinced anyone having lived under the Stasi rule of Eastern Germany will have some difficulty accepting to pledge behaving according to a feel-good CoC.

**Comment by TimKam** [comment]
@jfbu I think we all agree that the "pledging" wording is unfortunate. That's why I suggest using a CoC similar to the one Django has. I now added a CoC that is based on the one of Django, just a bit simpler.

**Review by stephenfin — APPROVED** [review]
This looks good to me. I'm still unsure how necessary it is, but if it helps some people and isn't a hindrance to me, I don't really care 😊

File: doc/_templates/index.html
Comment (jfbu) [inline doc/_templates/index.html:125]: Hope not to sound too nitpicky but how is one suppose to adhere to the CoC concretely?

File: README.rst
Comment (jfbu) [inline README.rst:93]: But how will user/contributor concretely adhere?

File: CODE_OF_CONDUCT
Comment (jfbu) [inline CODE_OF_CONDUCT:5]: How is this suppose to proceed?

**Review by jfbu — COMMENTED** [review]
As expressed already I will not strongly oppose. The text is better than one of original PR. However, it appears to me to express a very strong cultural bias influenced by current activism (you can call it social progress if you prefer) in occidental societies. And that it reflects habits from the workplace. It inscribes itself in these kind of things you learn to accept in your professional life as being part of "belonging to the culture of the firm". It is to my taste, a very artificial approach, which for example does not mention the abysmal difference in income between well-paid Google employees and third world farmers. It is a "feel-good" text as I have said already. It reminds me of eager adoption of so-called "inclusive" orthography which in French language is not at all like in English and leads to absurd rules. Many people simply follow suit, because the head of human resources has decided to use this vocabulary so that the company appears trendy enough. It is a bit sad that such an evolution would not originate from Sphinx community itself but be the result of a self-recognized zelot of such CoCs in software projects. I am not too happy about it but will not formally oppose it.

**Comment by jfbu** [comment]
> ut be the result of a self-recognized zelot of such CoCs in software projects.

of course I am not referring here to @TimKam  but to @tapaswenipathak. And apologies for spelling mistake.

**Comment by TimKam** [comment]
@jfbu thanks for your comments on this update.

1. People adhere to the CoC as long as they follow the rules, which can be colloquially summarized as "be nice and considerate, strive for consensus and compromise". We no longer claim we have the duty and power to enforce "hard" rules, we just ask.

2. I think we all agree that the big privilege divide will not be bridged by CoCs and inclusive language policies.

3. I agree that in the corporate world, similar practices are sometimes introduced with a purely opportunistic motivation.

4. However, I *do* think that a reminder to be nice can't hurt in an industry, in which some organizations have adopted an investment banking-like culture; I think these cultural issues do **not** fully translate to the Open Source world. But sometimes, weird things happen. Like [this Python project](https://pdoc3.github.io/pdoc/doc/pdoc/) showing a Swastika in the footer of its documentation (not claiming that *that* guy's attitude will be changed by a CoC, though).

5. I do not have a strong opinion, either, and am happy to work in further feedback. If no further concerns are raised, I will merge in ~a week if @tk0miya approves.

**Comment by TimKam** [comment]
@tk0miya Are you okay with merging this?

**Review by tk0miya — COMMENTED** [review]
LGTM!

@TimKam Thank you for great work!

File: README.rst
Comment (tk0miya) [reply README.rst:93]: How about adding a hyperlink to issue template?

Comment (TimKam) [reply README.rst:93]: Makes sense. I think you have the necessary permissions on GitHub (I don't have them).

Comment (tk0miya) [reply README.rst:93]: No, issue templates are managed under `.github/ISSUE_TEMPLATES/`. So you can modify them all :-)

**Comment by techtonik** [comment]
@TimKam to treat [the study](https://onlinelibrary.wiley.com/doi/abs/10.1002/%28SICI%291520-6793%28199910%2916%3A7%3C545%3A%3AAID-MAR1%3E3.0.CO%3B2-I?casa_token=G3QrgJqqLLwAAAAA%3ACjcjPsnWNoDe1ixz7bgULxbbrLQNbTEf62iz1lY_ZVNpq3pSsp8lMui_bxBj1JfurGtpr8893Eh8HjVR) that you've referenced fairly in the context of this issue. We need to replace "price negotiation" with "seller fixed price". Then the model becomes "take it or leave it" I doubt that priming fairness continues to have any value for the buyer of the "socialization service".

**Comment by jfbu** [comment]
I am adding here this [link](https://web.archive.org/web/20191022234432/https://meta.stackexchange.com/questions/334900/official-faq-on-gender-pronouns-and-code-of-conduct-changes) to document the kind of turmoil some types of CoC can lead into. It is to be hoped the Sphinx community, which handles matters of programming with no limitations on geographical origins, avoids getting as StackOverflow so much embroiled into agenda of activism much limited to English language and specific socio-cultural groups.

For commodity of those reading this, I am adding a [link](https://meta.stackexchange.com/questions/334399/summing-up-the-main-issues-the-story-so-far/334417#334417) which gives more insight into the events at SO.

Naturally, this is currently not directly relevant to the recent Sphinx community adoption of a CoC  but it points to the kind of problems which could arise in future and I think we should already now think about it in advance to avoid escalation  (I do not mean that a discussion should be held but simply that each one person potentially concerned should think about the issues at stake). As many commenters point out, it is a bit hard to understand how places where one discusses software issues end up being thrown into a state of chaos due to intrusion a very specific type (I won't dare say "gender") of human issues (whose legitimacy I absolutely do not disqualify, to the extent that they remain about achieving the freedom of each limited by the freedom of others).

**Pull request merged**
